src/mlpack/core/data/types.hpp

```diff
@@ -0,0 +1,60 @@
+/**
+ * @file core/data/types.hpp
+ * @author Gopi M. Tatiraju
+ * 
+ * This file contains utilitiy fucntions related to types of data.
+ * We have adapted all the standard types which are available in armadillo.
+ * 
+ * This file also contains functions to convery external file types to mlpack
+ * file types. In future if we the user of mlpack needs support of an external
+ * linear algebra library like armadillo, all fucntions related to handling the
+ * types goes here.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_CORE_DATA_TYPES_HPP
+#define MLPACK_CORE_DATA_TYPES_HPP
+
+#include <iostream>
+#include <limits>
+
+namespace mlpack
+{
+namespace data
+{
+
+enum struct file_type
```

Comment:
We typically name structs and classes in mlpack with camelcase, so I know it is different than Armadillo's convention, but can we call this `FileType` instead?